src/legacy/core_plugins/data/public/plugin.ts

```diff
@@ -18,78 +18,21 @@
  */
 
 import { CoreSetup, CoreStart, Plugin } from 'kibana/public';
-import {
-  DataPublicPluginStart,
-  addSearchStrategy,
-  defaultSearchStrategy,
-  DataPublicPluginSetup,
-} from '../../../../plugins/data/public';
-import { ExpressionsSetup } from '../../../../plugins/expressions/public';
-
-import {
-  setIndexPatterns,
-  setQueryService,
-  setUiSettings,
-  setInjectedMetadata,
-  setFieldFormats,
-  setSearchService,
-  setOverlays,
-  // eslint-disable-next-line @kbn/eslint/no-restricted-paths
-} from '../../../../plugins/data/public/services';
-import { setSearchServiceShim } from './services';
-import {
-  selectRangeAction,
-  SelectRangeActionContext,
-  ACTION_SELECT_RANGE,
-} from './actions/select_range_action';
-import {
-  valueClickAction,
-  ACTION_VALUE_CLICK,
-  ValueClickActionContext,
-} from './actions/value_click_action';
-import {
-  SELECT_RANGE_TRIGGER,
-  VALUE_CLICK_TRIGGER,
-  // eslint-disable-next-line @kbn/eslint/no-restricted-paths
-} from '../../../../plugins/embeddable/public/lib/triggers';
-import { UiActionsSetup, UiActionsStart } from '../../../../plugins/ui_actions/public';
-
-import { SearchSetup, SearchStart, SearchService } from './search/search_service';
-
-export interface DataPluginSetupDependencies {
-  data: DataPublicPluginSetup;
-  expressions: ExpressionsSetup;
-  uiActions: UiActionsSetup;
-}
-
-export interface DataPluginStartDependencies {
-  data: DataPublicPluginStart;
-  uiActions: UiActionsStart;
-}
+import { addSearchStrategy, defaultSearchStrategy } from '../../../../plugins/data/public';
 
 /**
  * Interface for this plugin's returned `setup` contract.
  *
  * @public
  */
-export interface DataSetup {
-  search: SearchSetup;
-}
+export interface DataSetup {} // eslint-disable-line @typescript-eslint/no-empty-interface
```

Comment:
Kept these empty interfaces in place because there were a few areas in Kibana still using them. (trying to keep things as small as possible).